app/src/main/java/org/p2p/wallet/auth/common/WalletWeb3AuthManager.kt

```diff
@@ -93,15 +94,28 @@ class WalletWeb3AuthManager(
         )
     }
 
+    // TODO remove after QA check screens!
+    private fun saveDeviceShareMock(deviceShare: String) {
+        secureStorage.saveString(KEY_DEVICE_SHARE, deviceShare)
+    }
+
     fun saveDeviceShare(deviceShare: String) {
-        secureStorage.saveString("${KEY_DEVICE_SHARE}_$lastUserId", deviceShare)
+        val share = parseDeviceShare(deviceShare)
+        share?.let {
+            it.userId = lastUserId
+            secureStorage.saveString(KEY_DEVICE_SHARE, gson.toJson(it))
+        }
+    }
+
+    private fun parseDeviceShare(deviceShare: String): DeviceShareKey? {
+        return gson.fromJson(deviceShare, DeviceShareKey::class.java)
     }
 
-    private fun hasDeviceShare(): Boolean = with(sharedPreferences) {
-        contains("${KEY_DEVICE_SHARE}_$lastUserId")
+    fun hasDeviceShare(): Boolean = with(sharedPreferences) {
+        contains(KEY_DEVICE_SHARE)
     }
 
-    private fun getDeviceShare(): String = secureStorage.getString("${KEY_DEVICE_SHARE}_$lastUserId").orEmpty()
+    private fun getDeviceShare(): String = secureStorage.getString(KEY_DEVICE_SHARE).orEmpty()
```

Comment:
are you sure we want to return empty string in case of absent device share?
I think it's okay to return `null` if no device share is found

Reply:
oh yes I think you are right

Reply:
and do we need to manipulate raw string here? i think it's okay to return the whole entity from gson right here

Reply:
yeah we can do that